packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3159,9 +3171,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
         && depth != null
         && _active
         && Widget.canUpdate(widget, newWidget));
+    // This method supposed to clear out the slots of forgotten children. We
```

Comment:
Maybe phrase this as: "Element was told to update and we can now release all the global key reservations of forgotten children. We cannot do this earlier because if the Element never updates, the forgotten children still represent global key duplications (the forgotten children are not removed from the tree until the call to update happens)."